.github/workflows/updates.yaml

```diff
@@ -158,63 +162,153 @@ jobs:
+            # template repository 
+            ###########
+            # 1. Make sure the template repo exists 
+            - name: Step 1. Verify the template repo exists
+              id: template-repo-exists
+              run: | 
+                # echo "::set-output name=slack_runurl::https://github.com/${REPOOWNER}/template-builder/actions/runs/${GITHUB_RUN_ID}"
+                # Why curl instead of the gh cli? Because on not found, the gh cli will return an exit code of 1 which immediately fails the step and the job
+                repoData=$(curl -s -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/repos/${TEMPLATEOWNER}/${{ matrix.template }})
+                repoID=$(echo "$repoData" | jq '.id')
+                echo "::notice::Repo id - ${repoID}"
+                repoFound=$(echo "$repoData" | jq '.message')
+                echo "::notice::Repo Found status - ${repoFound}"
+                if [[ -z ${repoID} || "${repoFound}" == '"Not Found"'  ]]; then
+                  message="The template repo for ${{ matrix.template }} doesn't exist and will need to be created before I can help you."
+                  echo "::error::${message}"
+                  echo "SLACK_MSG=${message}" >> $GITHUB_ENV
+                  echo "SLACK_PRURL=no-pr" >> $GITHUB_ENV
+                  echo "SLACK_STATUS=skip" >> $GITHUB_ENV
+                  exit 11;
+                fi
+            # 2. Verify the repository is set for automerge
+            - name: Step 2. Check if auto-merge is enabled for repository
+              run: |
+                autoMREnabled=$(gh api /repos/${TEMPLATEOWNER}/${{ matrix.template }} --jq '.allow_auto_merge')
+                if [[ "${autoMREnabled}" != "true" ]]; then
+                  gh api -X PATCH /repos/${TEMPLATEOWNER}/${{ matrix.template }} --field allow_auto_merge=true
+                fi  
+
+            #3. Get the template repo's default branch
+            - name: Step 3. Get template default branch
+              id: defaultbranch
+              run: |
+                # Get the default branch.
+                DEFAULT_BRANCH=$(gh api /repos/${TEMPLATEOWNER}/${{ matrix.template }} --jq '.default_branch')
+                echo "Our default branch for this template is ${DEFAULT_BRANCH}"
+                echo "::set-output name=branch::$DEFAULT_BRANCH"
+
+            # 4. Check if default branch has branch rules in place
+            # @todo do we bother to check, or do we just assume it isnt set and attempt to create it anyway? If we attempt to create it and it already
+            # exists, it should just return the same values anyway
+            - name: Step 4. Add branch protection rules to default branch
+              run: |
+                # @todo it would be nice if there was some way to verify if the protection rule we need is set and in place instead of always 
```

Comment:
If this helps https://gist.github.com/chadwcarlson/1938d8619ebe7338e91d720b06685a26 @gilzow

Reply:
oh yeah, i need to go remove that comment. Will include it an upcoming push